stepsfunction/handler.py

```diff
@@ -163,19 +178,186 @@ def ocr_page(event, context):
     See Callback Pattern at https://console.aws.amazon.com/states/home?region=us-east-1#/sampleProjects
     """
     LOG.info(f'event: {dumps(event)}')
-    activity_arn = os.environ.get('ACTIVITY_ARN', "")
+    LOG.info(f'boto3 version: {boto3.__version__} botocore version: {botocore.__version__}')
+
     s3rec = event['Records'][0]['s3']  # only the first, but there should only be one for S3
     bucket = s3rec['bucket']['name']
     key = s3rec['object']['key']
     _doc_pdf, jid, name_pdf = key.split('/')
     LOG.info(f'bucket={bucket} key={key} jid={jid} name_pdf={name_pdf}')
 
-    worker_name = "notify_upload_worker"
-    sf = boto3.client('stepfunctions')
+    detected_text = get_textract_data(bucket, key)
+    write_textract_to_s3(detected_text, bucket, key)
+    pdf_upload = PDFUpload.get(hash_key=jid)
+    update_db_after_ocr(pdf_upload, detected_text, name_pdf, jid)
```

Comment:
In future we probably will not want to save the extracted text to DDB, becuase the more data stored, the slower the retrieval is, and the higher the cost.  If we can't determine by S3 pattern (e.g., s3:BUCKETNAME/JID/page_text/0000.txt, then it's probably better to put the S3 key in the DDB, instead of the text.